Code/Framework/AzCore/Platform/iOS/AzCore/NativeUI/NativeUISystemComponent_iOS.mm

```diff
@@ -31,6 +31,61 @@
             
             NSString* nsTitle = [NSString stringWithUTF8String:title.c_str()];
             NSString* nsMessage = [NSString stringWithUTF8String:message.c_str()];
+            
+#if defined(CARBONATED)
+            void (^DisplayBlockingDialogCommon)(void) = ^{
+                UIAlertController* alert = [UIAlertController alertControllerWithTitle:nsTitle message:nsMessage preferredStyle:UIAlertControllerStyleAlert];
+                
+                for (int i = 0; i < options.size(); i++)
+                {
+                    UIAlertAction* okAction = [UIAlertAction actionWithTitle:[NSString stringWithUTF8String:options[i].c_str()] style:UIAlertActionStyleDefault handler:^(UIAlertAction* action) { userSelection = [action.title UTF8String]; }];
+                    [alert addAction:okAction];
+                }
+                
+                UIWindow* foundWindow = nil;
+#if defined(__IPHONE_13_0) || defined(__TVOS_13_0)
+                if(@available(iOS 13.0, tvOS 13.0, *))
+                {
+                    NSArray* windows = [[UIApplication sharedApplication] windows];
+                    for (UIWindow* window in windows)
+                    {
+                        if (window.isKeyWindow)
+                        {
+                            foundWindow = window;
+                            break;
+                        }
+                    }
+                }
+#else
+                foundWindow = [[UIApplication sharedApplication] keyWindow];
+#endif
+                UIViewController* rootViewController = foundWindow ? foundWindow.rootViewController : nullptr;
+                if (rootViewController)
```

Comment:
Please simulate missing window (so there is no root controller), I guess there'll be a deadlock because no signal. I believe the display call should make an action to release the waiting thread in this case.